source/blog/2016-03-13-ember-data-2-4-released.md

```diff
@@ -0,0 +1,169 @@
+
+```js
+// GET /articles/1?include=comments
+
+var article = this.store.findRecord('article', 1, { include: 'comments' });
+```
+
+```js
+// GET /articles?include=comments
+
+var article = this.store.findAll('article', { include: 'comments' });
+```
+
+#### `ds-references`
+
+The `ds-references` feature implements the references API as described
+in [RFC 57](https://github.com/emberjs/rfcs/pull/57). References is a
+low level API to perform meta-operations on records, has-many
+relationships and belongs-to relationships:
+
+  * get the current local data synchronously without triggering a fetch or producing a promise
+  * notify the store that a fetch for a given record has begun, and provide a promise for its result
+  * similarly, notify a record that a fetch for a given relationship has begun, and provide a promise for its result
+  * retrieve server-provided metadata about a record or relationship
+
+Consider the following `post` model:
+
+```js
+// app/models/post.js
+import Model from 'ember-data/model';
+import { belongsTo, hasMany } from 'ember-data/relationships';
+
+export default Model.extend({
+  comments: hasMany(),
+  author: belongsTo()
+});
+```
+
+The references API now allows the possibility to interact with the relationships:
+
+```js
+var post = store.peekRecord('post', 1);
+
+// check if the author is already loaded, without triggering a request
+if (post.belongsTo('author').value() !== null) {
+  console.log(post.get("author.name"));
+} else {
+  // load the author
+  post.belongsTo('author').load();
+}
+
+// reload the author
+post.belongsTo('author').reload();
+
+// check if there are comments, without triggering a request
+if (post.hasMany('comments').value() !== null) {
+  var ids = post.hasMany('comments').ids();
+
+  var meta = post.hasMany('comments').meta();
+  console.log(`${ids.length} comments out of ${meta.total}`);
+} else {
+  post.hasMany('comments').load();
+}
+
+// reload comments
+post.hasMany('comments').reload();
+```
+
+Thanks to [@pangratz](https://github.com/pangratz) for implementing
+this feature.
+
+
+#### `ds-transform-pass-options
+
+The `ds-transform-pass-options` feature allows for smarter transforms
+by passing the options object form `DS.attr([type], [options])` into
+the transform.
+
+
+##### Example
+```app/models/post.js
+export default DS.Model.extend({
+  title: DS.attr('string'),
+  markdown: DS.attr('markdown', {
+    markdown: {
+      gfm: false,
+      sanitize: true
+    }
+  })
+});
+```
+
+```app/transforms/markdown.js
+export default DS.Transform.extend({
+  serialize: function (deserialized, attributeMeta) {
+    return deserialized.raw;
+  },
+
+  deserialize: function (serialized, attributeMeta) {
+    var options = attributeMeta.options.markdown || {};
+
+    return marked(serialized, options);
+  }
+});
+```
+
+Thanks to [@pangratz](https://github.com/pangratz) for implementing
+this feature and [@knownasilya](https://github.com/knownasilya) for
+proposing the RFC.
```

Comment:
Tiny nitpick: I think for the sake of consistency the RFC should be linked too https://github.com/emberjs/rfcs/pull/1...